src/sentry/tasks/reports.py

```diff
@@ -270,15 +319,59 @@ def prepare_project_issue_summaries(interval, project):
     return [new_issue_count, reopened_issue_count, existing_issue_count]
 
 
-def prepare_project_usage_summary(start__stop, project):
+def prepare_project_usage_outcomes(start__stop, project):
     start, stop = start__stop
+
+    # XXX(epurkhiser): Tsdb used to use day buckets, where the end would
+    # represent a whole day. Snuba queries more accurately thus we must
+    # capture the entire last day
+    end = stop + timedelta(days=1)
+
+    query = Query(
+        dataset=Dataset.Outcomes.value,
+        match=Entity("outcomes"),
+        select=[
+            Column("outcome"),
+            Column("category"),
+            Function("sum", [Column("quantity")], "total"),
+        ],
+        where=[
+            Condition(Column("timestamp"), Op.GTE, start),
+            Condition(Column("timestamp"), Op.LT, end),
+            Condition(Column("project_id"), Op.EQ, project.id),
+            Condition(Column("org_id"), Op.EQ, project.organization_id),
+            Condition(
+                Column("outcome"), Op.IN, [Outcome.ACCEPTED, Outcome.FILTERED, Outcome.RATE_LIMITED]
+            ),
+            Condition(
+                Column("category"),
+                Op.IN,
+                [*DataCategory.error_categories(), DataCategory.TRANSACTION],
+            ),
+        ],
+        groupby=[Column("outcome"), Column("category")],
+        granularity=Granularity(ONE_DAY),
+    )
+    data = raw_snql_query(query, referrer="reports.outcomes")["data"]
+
     return (
-        tsdb.get_sums(
-            tsdb.models.project_total_blacklisted, [project.id], start, stop, rollup=60 * 60 * 24
-        )[project.id],
-        tsdb.get_sums(
-            tsdb.models.project_total_rejected, [project.id], start, stop, rollup=60 * 60 * 24
-        )[project.id],
+        # accepted transactions
+        sum(
+            row["total"]
+            for row in data
+            if row["category"] == DataCategory.TRANSACTION and row["outcome"] == Outcome.ACCEPTED
+        ),
+        # Accepted errors
+        sum(
+            row["total"]
+            for row in data
+            if row["category"] in DataCategory.error_categories()
+            and row["outcome"] == Outcome.ACCEPTED
+        ),
+        # Filtered
+        sum(row["total"] for row in data if row["outcome"] == Outcome.FILTERED),
+        # Rate limited
+        sum(row["total"] for row in data if row["outcome"] == Outcome.RATE_LIMITED),
```

Comment:
These totals will include filtered/rate limited transactions. Do you need to separate results by event type?

Reply:
Yeah, there's an ongoing discussion around this.